sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py

```diff
@@ -2628,3 +2628,61 @@ def register_both_output():
         node_output = pipeline_job.jobs['node'].outputs.component_out_path
         assert node_output.name == 'a_output'
         assert node_output.version == '1'
+
+    def test_register_output_sdk_succeed(self, client: MLClient, randstr: Callable[[str], str], randint: Callable[[int], int]):
+        component = load_component(source="./tests/test_configs/components/helloworld_component.yml")
+        component_input = Input(type='uri_file', path='https://dprepdata.blob.core.windows.net/demo/Titanic.csv')
+
+        random_version = randstr("version")
+
+        @dsl.pipeline()
+        def sub_pipeline():
+            node = component(component_in_path=component_input)
+            node.outputs.component_out_path.name = 'sub_pipeline_output'
+            node.outputs.component_out_path.version = random_version
+            return {
+                'sub_pipeine_a_output': node.outputs.component_out_path
+            }
+
+        @dsl.pipeline()
+        def register_both_output():
+            # register NodeOutput which is binding to PipelineOutput
+            node = component(component_in_path=component_input)
+            node.outputs.component_out_path.name = 'n1_output'
+            node.outputs.component_out_path.version = random_version
+
+            # register NodeOutput which isn't binding to PipelineOutput
+            node_2 = component(component_in_path=component_input)
+            node_2.outputs.component_out_path.name = 'n2_output'
+            node_2.outputs.component_out_path.version = random_version
+
+            # register NodeOutput of subgraph
+            sub_node = sub_pipeline()
+            sub_node.outputs.sub_pipeine_a_output.name = 'sub_pipeline'
+            sub_node.outputs.sub_pipeine_a_output.version = random_version
+
+            return {
+                'pipeine_a_output': node.outputs.component_out_path
+            }
+
+        pipeline = register_both_output()
+        pipeline.outputs.pipeine_a_output.name = 'p1_output'
+        pipeline.outputs.pipeine_a_output.version = random_version
+        pipeline.settings.default_compute = "cpu-cluster"
+        pipeline_job = client.jobs.create_or_update(pipeline)
+        client.jobs.stream(pipeline_job.name)
+
+        pipeline_output = pipeline_job.outputs.pipeine_a_output
+        assert pipeline_output.name == 'p1_output'
+        assert pipeline_output.version == random_version
+        assert client.data.get(name='p1_output', version=random_version)
+
+        node_output = pipeline_job.jobs['node_2'].outputs.component_out_path
+        assert node_output.name == 'n2_output'
+        assert node_output.version == random_version
+        assert client.data.get(name='n2_output', version=random_version)
+
+        node_output = pipeline_job.jobs['sub_node'].outputs.sub_pipeine_a_output
```

Comment:
we can also have a extract function here: check_name_version_and_register_succeed